File: LeetCode/0027.remove-element/remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int size = nums.size();
        int move = 0;
        for (int i = 0; i < size; i++){
            nums[i - move] = nums[i];
            if (nums[i] == val) move++;
        }
        return size - move;
    }
};

File: LeetCode/0027.remove-element/remove-element.kt
class Solution {
    fun removeElement(nums: IntArray, `val`: Int): Int {
        var move = 0
        for (index in nums.indices) {
            nums[index - move] = nums[index]
            if (nums[index] == `val`) {
                move++
            }
        }
        return nums.size - move
    }
}